src/coreclr/nativeaot/System.Private.StackTraceMetadata/src/Internal/StackTraceMetadata/StackTraceMetadata.cs

```diff
@@ -41,6 +42,10 @@ internal static void Initialize()
             RuntimeAugments.InitializeStackTraceMetadataSupport(new StackTraceMetadataCallbacksImpl());
         }
 
+        [Intrinsic]
+        [AnalysisCharacteristic]
+        internal static extern bool StackTraceHiddenMetadataPresent();
```

Comment:
It looks like this mechanism relies on us always optimizing out the call to this method. Where do we generate an IL body for it in the no-scanner case? Should we just provide a default body of `=> true` here?

Reply:
SubstitutedILProvider does this - we treat this body as always substituted. If there's no whole program view (characteristics are unknown), it substitutes true, otherwise it substitutes as needed.